CMakeLists.txt

```diff
@@ -7,16 +7,19 @@ if(POLICY CMP0074)
   cmake_policy(SET CMP0074 NEW)
 endif()
 
-# The native FindLibArchive is rather naive.  This allows overruling
-# the libarchive dependencies
-if(LibArchive_LIBRARIES AND LibArchive_INCLUDE_DIRS)
-  set(LibArchive_FOUND ON)
+find_package(LibArchive)
+
+if(LibArchive_FOUND)
+
+if(NOT LibArchive_OS_LIBS)
+if(WIN32)
+  set(LibArchive_OS_LIBS nettle.lib xml2.lib iconv.lib regex.lib expat.lib lzma.lib zstd.lib lz4.lib bz2.lib z.lib bcrypt.lib ws2_32.lib)
```

Comment:
Hmm.  This is not needed (and undesirable) for the DLL version.   Another problem is that the set of libraries depends on the included formats and filters in libarchive.    I fear we need a more dynamic way to get this list 😢